cloud/grpc.go

```diff
@@ -27,18 +44,21 @@ func (c *client) UnaryAuthInterceptor() grpc.UnaryClientInterceptor {
 		err := invoker(ctx, method, req, reply, cc, opts...)
 		if err != nil {
 			s, ok := status.FromError(err)
-			if ok && s.Code() == codes.Unauthenticated {
-				err = c.Authenticate(ctx)
-				if err != nil {
-					return errors.Wrap(err, "failed re-authenticating")
+			if ok {
+				err = status.Errorf(s.Code(), fmt.Sprintf("%s {reqID: %s}", s.Err(), getReqID(ctx)))
+				if s.Code() == codes.Unauthenticated {
+					err = c.Authenticate(ctx)
+					if err != nil {
+						return errors.Wrap(err, "failed re-authenticating")
+					}
+					md, ok := metadata.FromOutgoingContext(ctx)
+					if !ok {
+						return errors.New("could not parse metadata")
```

Comment:
Should this really be an error - couldn't we just start a new metadata object if it doesn't exist?

Reply:
I've updated this to start new metadata as you suggested